packages/core/src/sharedWishlists/redux/selectors.js

```diff
@@ -0,0 +1,151 @@
+ * @param {object} state - Application state.
+ *
+ * @returns {string} Universal idenfier of the shared wishlist.
+ *
+ * @example
+ * import { getSharedWishlistId } from '@farfetch/blackout-redux';
+ *
+ * const mapStateToProps = state => ({
+ *    sharedWishlistId: getSharedWishlist(state)
+ * });
+ */
+export const getSharedWishlistId = state =>
+  fromSharedWishlistReducer.getResult(state.sharedWishlist);
+
+/**
+ * Retrieves the error state of the current user's shared wishlist.
+ *
+ * @function
+ * @memberof module:sharedWishlists/selectors
+ *
+ * @param {object} state - Application state.
+ *
+ * @returns {object|undefined} Error information, `undefined` if there are no errors.
+ *
+ * @example
+ * import { getSharedWishlistError } from '@farfetch/blackout-redux';
+ *
+ * const mapStateToProps = state => ({
+ *    error: getSharedWishlistError(state)
+ * });
+ */
+export const getSharedWishlistError = state =>
+  fromSharedWishlistReducer.getError(state.sharedWishlist);
+
+/**
+ * Retrieves the loading status of the shared wishlist.
+ *
+ * @function
+ * @memberof module:sharedWishlists/selectors
+ *
+ * @param {object} state - Application state.
+ *
+ * @returns {boolean} Loading status of the shared wishlist.
+ *
+ * @example
+ * import { getSharedWishlistLoading } from '@farfetch/blackout-redux';
+ *
+ * const mapStateToProps = state => ({
+ *    isLoading: getSharedWishlistLoading(state)
+ * });
+ */
+export const getSharedWishlistLoading = state =>
+  fromSharedWishlistReducer.getIsLoading(state.sharedWishlist);
+
+/**
+ * Retrieves a specific shared wishlist by its id, with all properties populated(ie,
+ * the product).
+ *
+ * @function
+ * @memberof module:sharedWishlists/selectors
+ *
+ * @param {object} state - Application state.
+ * @param {string} sharedWishlistId - Numeric identifier of the shared wishlist.
+ *
+ * @returns {object} - Shared wishlist entity for the given id.
+ *
+ * @example
+ * import { getSharedWishlist } from '@farfetch/blackout-redux';
+ *
+ * const mapStateToProps = (state, { sharedWishlists: { id } }) => ({
+ *    sharedWishlist: getSharedWishlist(state, id)
+ * });
+ */
+export const getSharedWishlist = (state, sharedWishlistId) => {
+  return getEntity(state, 'sharedWishlists', sharedWishlistId);
+};
+
+/**
+ * Retrieves a specific shared wishlist item by its id, with all properties
+ * populated (ie, the product).
+ *
+ * @function
+ * @memberof module:sharedWishlists/selectors
+ *
+ * @param {object} state - Application state.
+ * @param {number} sharedWishlistItemId - Numeric identifier of the shared wishlist item.
+ *
+ * @returns {object} - Shared wishlist item entity for the given id.
+ *
+ * @example
+ * import { getSharedWishlistItem } from '@farfetch/blackout-redux';
+ *
+ * const mapStateToProps = (state, { sharedWishlistItem: { id } }) => ({
+ *    sharedWishlistItem: getSharedWishlistItem(state, id)
+ * });
+ */
+export const getSharedWishlistItem = (state, sharedWishlistItemId) => {
```

Comment:
This should use a `createSelector` as it will be creating a new reference each time. But then, you will need to change the `getSharedWishlistItems` selector to avoid that call from invalidating the cache inside the loop. Check the implementation in v2 and do the same please.

Reply:
Implemented as in `v2`!